primitives/arithmetic/src/traits.rs

```diff
@@ -145,7 +145,15 @@ impl<T: Clone + Zero + One + PartialOrd + CheckedMul + Bounded + num_traits::Sat
 	}
 
 	fn saturating_pow(self, exp: usize) -> Self {
-		checked_pow(self, exp).unwrap_or_else(Bounded::max_value)
+		let neg = self < T::zero() && exp % 2 != 0;
```

Comment:
this this and checked_pow handle `n^0 == 1`?

Reply:
Yep, I added some tests for `Saturating` trait. It could be obvious so I can remove them if you ask.